test/balanced-pool.js

```diff
@@ -163,3 +163,204 @@ test('busy', (t) => {
     }
   })
 })
+
+const cases = [ 
+  // 0
+  { 
+    iterations: 25, 
+    servers: [{ name: 'A', disabled:[0] }, { name: 'B' }, { name: 'C' }], 
```

Comment:
@mcollina  disabled[0] means that server A will be disabled only in iteration number 0 and it will be enabled again after 0. I'll have to think of a way to make it understandable or I might add comments

Reply:
`iterations` is the number of requests.  maybe I should rename that to requests 🤔

Reply:
@mcollina Can you have a look and tell me what do you think of the tests before I start cleaning?